algolia/vue-instantsearch
# feat(server side rendering): support ais-index

## What?

- Updated the API for server side rendering 
- Update internal APIs (forceRender) to take index widgets in account
- Allow InstantSearch to receive previously searched results
- Updated the examples as well with the local version of Vue InstantSearch.
- small changes to the Netlify & e2e test configuration to optimise them

## How?

Rendering once in a provided method (findResultsState) which gets used in `serverPretch` (a Vue feature) to add all widgets due to the regular "created" hook. Then we search, like InstantSearch does as well (using `searchOnlyDerivedHelpers`) and use those results to do the actual rendering of the app. Frontend-wise this uses those results too, so there's an API called `hydrate` (like before) which takes those results and sets them on a place InstantSearch expects in forceRender. There's also `getState` to allow someone to get access to the results outside of the `serverPrefetch` hook.

Fixes made to InstantSearch.js itself https://github.com/algolia/instantsearch.js/pull/4359 (keeping ui state in the right order even with different widget mounting order)

## Notes

BREAKING CHANGE: `createInstantSearch` is now `createServerRootMixin`, which should be used at the highest up level. See the examples for more info.

## How to test this?

1. check out this branch
2. `yarn build` (can comment out non-esm parts in the rollup config)
3. open the server side rendering or nuxt example
